lib/class-wp-rest-meta-fields.php

```diff
@@ -0,0 +1,291 @@
+	 * @return bool|WP_Error True if meta fields are updated, error otherwise.
+	 */
+	protected function update_multi_meta_value( $object, $name, $values ) {
+		if ( ! current_user_can( 'edit_post_meta', $object, $name ) ) {
+			return new WP_Error(
+				'rest_cannot_update',
+				sprintf( __( 'You do not have permission to edit the %s custom field.' ), $name ),
+				array( 'key' => $name, 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		$current = get_metadata( $this->get_meta_type(), $object, $name, false );
+		$to_add = array_diff( $values, $current );
+		$to_remove = array_diff( $current, $values );
+
+		foreach ( $to_add as $value ) {
+			if ( ! add_metadata( $this->get_meta_type(), $object, wp_slash( $name ), wp_slash( $value ) ) ) {
+				return new WP_Error(
+					'rest_meta_database_error',
+					__( 'Could not update meta value in database.' ),
+					array( 'key' => $name, 'status' => WP_HTTP::INTERNAL_SERVER_ERROR )
+				);
+			}
+		}
+		foreach ( $to_remove as $value ) {
+			if ( ! delete_metadata( $this->get_meta_type(), $object, wp_slash( $name ), wp_slash( $value ) ) ) {
+				return new WP_Error(
+					'rest_meta_database_error',
+					__( 'Could not update meta value in database.' ),
+					array( 'key' => $name, 'status' => WP_HTTP::INTERNAL_SERVER_ERROR )
+				);
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Update meta value for an object.
+	 *
+	 * @param int $object Object ID.
+	 * @param string $name Key for the custom field.
+	 * @return bool|WP_Error True if meta field is deleted, error otherwise.
+	 */
+	protected function update_meta_value( $object, $name, $value ) {
+		if ( ! current_user_can( 'edit_post_meta', $object, $name ) ) {
+			return new WP_Error(
+				'rest_cannot_update',
+				sprintf( __( 'You do not have permission to edit the %s custom field.' ), $name ),
+				array( 'key' => $name, 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		if ( ! update_metadata( $this->get_meta_type(), $object, wp_slash( $name ), wp_slash( $value ) ) ) {
+			return new WP_Error(
+				'rest_meta_database_error',
+				__( 'Could not update meta value in database.' ),
+				array( 'key' => $name, 'status' => WP_HTTP::INTERNAL_SERVER_ERROR )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get all the registered options for the Settings API
+	 *
+	 * @return array
+	 */
+	protected function get_registered_fields() {
+		$rest_options = array();
+
+		foreach ( get_registered_meta_keys( $this->get_meta_type() ) as $name => $args ) {
+			if ( empty( $args['show_in_rest'] ) ) {
+				continue;
+			}
+
+			$rest_args = array();
+			if ( is_array( $args['show_in_rest'] ) ) {
+				$rest_args = $args['show_in_rest'];
+			}
+
+			$default_args = array(
+				'name'             => $name,
+				'single'           => $args['single'],
+				'schema'           => array(),
+				'prepare_callback' => 'meta_rest_api_prepare_value',
+			);
+			$default_schema = array(
+				'type'        => null,
```

Comment:
It might be simpler to set the default type to `$args['type']`, as you wouldn't then need the double check below.